stdlib/CMakeLists.txt

```diff
@@ -18,6 +18,18 @@ if(SWIFT_BUILD_STATIC_STDLIB)
   list(APPEND SWIFT_STDLIB_LIBRARY_BUILD_TYPES STATIC)
 endif()
 
+# Also make a list of "stdlib extras".
+# This will determine if StdlibUnitTest, Platform/*,
+# SwiftPrivate, etc will build. 
+# This allows this logic to work with overlay only builds.
```

Comment:
"Make a list of "standard library extras" build types that the build should perform.

Or something like that. I think the comment as I wrote it really needs to be clear that you are selecting the specific things to build.